### PR TITLE
Improve null statistics handling in product tests

### PR DESCRIPTION
**Pull request by wendigo**
Fixes https://github.com/trinodb/trino/issues/10366

### CODE REVIEW

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/StatisticsFetcher.java [hunk 1]
```diff
@@ -90,30 +97,29 @@ private Stats toStats(Statistics statistics, Stats previousStats)
         }
 
         Stats stats = new Stats();
-        stats.systemCpuUsage = statistics.getCpuStats().getSystemCpuUsage();
-        stats.totalCpuUsage = statistics.getCpuStats().getCpuUsage().getTotalUsage();
+        stats.systemCpuUsage = firstNonNull(statistics.getCpuStats().getSystemCpuUsage(), -1L);
+        stats.totalCpuUsage = Optional.ofNullable(statistics.getCpuStats().getCpuUsage()).map(CpuUsageConfig::getTotalUsage).orElse(-1L);
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/StatisticsFetcher.java:101]: or 0 ?

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/StatisticsFetcher.java [hunk 2]
```diff
@@ -90,30 +97,29 @@ private Stats toStats(Statistics statistics, Stats previousStats)
...
         }
 
         MemoryStatsConfig memoryStats = statistics.getMemoryStats();
-        stats.memoryLimit = DataSize.ofBytes(memoryStats.getLimit()).to(GIGABYTE);
-        stats.memoryUsage = DataSize.ofBytes(memoryStats.getUsage()).to(GIGABYTE);
-        stats.memoryMaxUsage = DataSize.ofBytes(memoryStats.getMaxUsage()).to(GIGABYTE);
-        stats.memoryUsagePerc = 100.0 * memoryStats.getUsage() / memoryStats.getLimit();
+        stats.memoryLimit = DataSize.ofBytes(firstNonNull(memoryStats.getLimit(), 0L)).to(GIGABYTE);
+        stats.memoryUsage = DataSize.ofBytes(firstNonNull(memoryStats.getUsage(), 0L)).to(GIGABYTE);
+        stats.memoryUsagePerc = 100.0 * firstNonNull(memoryStats.getUsage(), 1L) / firstNonNull(memoryStats.getLimit(), 1L);
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/StatisticsFetcher.java:116]: why 1?

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/StatisticsFetcher.java [hunk 3]
```diff
@@ -191,7 +190,6 @@ public void close()
                 "mem",
                 "max mem",
                 "mem %",
-                "peak mem",
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/StatisticsFetcher.java:187]: > Remove peak memory usage statistic

in the cmt msg, include the why

**Review by findepi — APPROVED** [review]

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/StatisticsFetcher.java (hunk 1 above)
Comment (wendigo) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/StatisticsFetcher.java:101]: -1 so the cpu usage won't be calculated if there are no stats

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/StatisticsFetcher.java (hunk 2 above)
Comment (wendigo) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/StatisticsFetcher.java:116]: corrected.

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/StatisticsFetcher.java (hunk 3 above)
Comment (wendigo) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/StatisticsFetcher.java:187]: Done

**Review by findepi — APPROVED** [review]

**Comment by wendigo** [comment]
@findepi can we merge that?

**Comment by findepi** [comment]
it's red. What's the problem?

**Comment by wendigo** [comment]
Some unrelated flakyness of phoenix tests: 

`Caused by: org.apache.hbase.thirdparty.io.netty.channel.unix.Errors$NativeIoException: bind(..) failed: Cannot assign requested address`

**Comment by findepi** [comment]
Do we have an issue for this?

**Comment by wendigo** [comment]
> Do we have an issue for this?

I'll create one @findepi

**Comment by findepi** [comment]
> 
> I'll create one 

thanks!

**Pull request merged**